Sources/Routing/TrieRouter.swift

```diff
@@ -53,8 +57,12 @@ public final class TrieRouter<Output> {
                 case .constant(let data, let size):
                     // if we find a constant route path that matches this component,
                     // then we should use it.
-                    let match = component.withByteBuffer { buffer in
-                        return memcmp(data, buffer.baseAddress!, size) == 0
+                    let match = component.withByteBuffer { buffer -> Bool in
+                        if self.caseInsensitive {
+                            return data.caseInsensitiveEquals(to: buffer)
+                        } else {
+                            return memcmp(data, buffer.baseAddress!, size) == 0
```

Comment:
@gwynne is there a more ABI-safe way to do this?

Reply:
```swift
return data.elementsEqual(buffer)
```

Reply:
In addition to any safety issues, the use of `memcmp`currently causes a bug where routes with `A` and `AA` will always end up resolving to the `A` route.

Reply:
That's less `memcmp()`'s fault per se and more of an ordering logic error.

For the record, `memcmp()` is not really _unsafe_ in this particular usage so much as it is bad style; one should always use the Swift stdlib whenever possible, even when working with raw pointers.

Reply:
@gwynne you're the best! ty

Reply:
@kevinup7 I will add a test case for that. That's a bug